go-controller/pkg/cni/helper_linux.go

```diff
@@ -365,17 +365,41 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 	// Find and remove any existing OVS port with this iface-id. Pods can
 	// have multiple sandboxes if some are waiting for garbage collection,
 	// but only the latest one should have the iface-id set.
-	uuids, _ := ovsFind("Interface", "_uuid", "external-ids:iface-id="+ifaceID)
-	for _, uuid := range uuids {
-		if out, err := ovsExec("remove", "Interface", uuid, "external-ids", "iface-id"); err != nil {
-			klog.Warningf("Failed to clear stale OVS port %q iface-id %q: %v\n  %q", uuid, ifaceID, err, out)
+	names, _ := ovsFind("Interface", "name", "external-ids:iface-id="+ifaceID)
+	for _, name := range names {
+		if name == hostIfaceName {
+			// this may be result of restarting ovnkube-node, and it is trying to add the same VF representor to
+			// br-int for the same pod; do not delete port in this case.
+			continue
+		}
+		if out, err := ovsExec("--if-exists", "--with-iface", "del-port", "br-int", name); err != nil {
+			klog.Warningf("Failed to delete stale OVS port %q with iface-id %q from br-int: %v\n %q",
+				name, ifaceID, err, out)
+		}
+	}
+
+	// if the specified port was created for other Pod/NAD, return error
+	extIds, err := ovsFind("Interface", "external_ids", "name="+hostIfaceName)
+	if err == nil && len(extIds) == 1 {
+		extId := extIds[0]
+		ifaceIDStr := util.GetExternalIDValByKey(extId, "iface-id")
+		nadNameString := util.GetExternalIDValByKey(extId, types.NADExternalID)
+		// if NADExternalID does not exists, it is default network
+		if nadNameString == "" {
+			nadNameString = types.DefaultNetworkName
+		}
+		if ifaceIDStr != ifaceID {
+			return fmt.Errorf("OVS port %s was added for iface-id (%s), now readding it for (%s)", hostIfaceName, ifaceIDStr, ifaceID)
+		}
+		if nadNameString != ifInfo.NADName {
+			return fmt.Errorf("OVS port %s was added for NAD (%s), expect (%s)", hostIfaceName, nadNameString, ifInfo.NADName)
 		}
 	}
 
 	// Add the new sandbox's OVS port, tag the port as transient so stale
 	// pod ports are scrubbed on hard reboot
 	ovsArgs := []string{
-		"add-port", "br-int", hostIfaceName, "other_config:transient=true", "--", "set",
+		"--may-exist", "add-port", "br-int", hostIfaceName, "other_config:transient=true", "--", "set",
```

Comment:
@trozet We are changing the behavior of a common function. PTAL.

Reply:
`ConfigureOVS()` should be used outside of the DPU context

Reply:
@wizhaoredhat sorry, can you please clarify what do you mean configureOVS used outside of dpu context? What needs to be fixed here? Thanks!

Reply:
@cathy-zhou Sorry I wasn't too clear. Prior to your change here adding a port will result in an error. By adding `may-exist`, we no long error on this condition. I am assuming you added this because of some cases you have seen in DPU where this condition occurs, but we don't want to error out. I don't see any mention of any case in the commit message, it may help if you explain what you have seen in the past which prompted this change. I am not too sure if this will affect cases where the DPU is not used, where we do want to error out if the port already exists. That is why I asked Tim to take a look, because I am not sure if this will have any impact. (This comment applies to all the other places you added `may-exist`).

Reply:
sure. but I think --may-exist would be needed, otherwise, everytime when ovnkube-node-dpu restarts, this function would fail, right?